src/Components/Components/src/CascadingParameterState.cs

```diff
@@ -60,16 +60,17 @@ public static IReadOnlyList<CascadingParameterState> FindCascadingParameters(Com
 
     private static ICascadingValueComponent? GetMatchingCascadingValueSupplier(in ReflectedCascadingParameterInfo info, ComponentState componentState)
     {
+        var candidate = componentState;
         do
         {
-            if (componentState.Component is ICascadingValueComponent candidateSupplier
+            if (candidate.Component is ICascadingValueComponent candidateSupplier
                 && candidateSupplier.CanSupplyValue(info.ValueType, info.SupplierValueName))
             {
                 return candidateSupplier;
             }
 
-            componentState = componentState.ParentComponentState;
-        } while (componentState != null);
+            candidate = candidate.ParentComponentState;
+        } while (candidate != null);
```

Comment:
The changes here are purely to make it better reflect the nullability of `.ParentComponentState`. Previously it wasn't marked as nullable but as part of adding it to public API, I did mark it as nullable (because that's the truth). There should be no functional changes here though.